### PR TITLE
M5805 API Design Guidelines Documentation

### PR DESCRIPTION
**Pull request by tommydeboer**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits
- [ ] Added Feature/Fix to release notes
- [ ] Integration tests run correctly

### CODE REVIEW

File: docs/developer_documentation/api-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| Export API | `/api/export/v1` | `/export` |
+| Job API | `/api/job/v1/` | `/run`, `/jobs` |
+
+_Note: This naming convention hasn't been applied to any APIs yet, but will be in the future. For an overview 
+of available APIs, see the API Documentation plugin in a running MOLGENIS app._
+
+### Versioning
+Up the version when:
+- the response for existing endpoints has changed?
+- new endpoints are added?
```
Comment (mark-de-haan) [inline docs/developer_documentation/api-guidelines.md]: No, unless the following scenario's:
* A new endpoint replaces an existing one AND the endpoint is called differently
* A new endpoint affects the functionality of another endpoint (by taking over certain tasks?)

File: docs/developer_documentation/api-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+`/api/internal/<api_human_readable_identifier>/*`
+Internal APIs don't have a version because the corresponding plugins will evolve together with the API.
+
+For example:
+
+| API | URI | Endpoints |
+|:---|:---|:---------|
+| Authentication API | `/api/auth/v1` | `/login`, `/logout` |                    
+| REST API | `/api/rest/v1` | REST CRUD operations on resources and resource collections |
+| Metadata API | `/api/internal/meta/v1` | ... |
```
Comment (mark-de-haan) [inline docs/developer_documentation/api-guidelines.md]: unless the api/rest/ will expose metadata via an endpoint, I would like to have this as a public API

Comment (fdlk) [reply docs/developer_documentation/api-guidelines.md]: Would you agree that the current metadata editing api is still an internal api?

Comment (mark-de-haan) [reply docs/developer_documentation/api-guidelines.md]: Yes that is true, but if this is a list of "possible how we want it to be" endpoints we should keep this in mind

Comment (fdlk) [reply docs/developer_documentation/api-guidelines.md]: @ConnorStroomberg interesting! let's talk!

**Comment by tommydeboer** [comment]
Continued as google doc

**Pull request closed without merging**